x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -232,19 +235,136 @@ private static void validateField(Map<?, ?> properties, String fieldName, boolea
+                if (matchType.equals("date_range")) {
+                    List<String> formatEntries = collapseFieldMappingsValue(response, toFieldMappingProperty(matchField, "format"));
+                    if (formatEntries.isEmpty()) {
+                        createEnrichMappingBuilder(
+                            (builder) -> builder.field("type", matchType).field("doc_values", false),
+                            resultListener
+                        );
+                    } else if (formatEntries.size() > 1) {
+                        resultListener.onFailure(
+                            new ElasticsearchException(
+                                "Multiple distinct date format specified for match field '{}' - indices({})  format entries({})",
+                                policy.getMatchField(),
+                                Strings.collectionToCommaDelimitedString(policy.getIndices()),
+                                Strings.collectionToCommaDelimitedString(formatEntries)
+                            )
+                        );
+                    } else {
+                        createEnrichMappingBuilder(
+                            (builder) -> builder.field("type", matchType).field("doc_values", false).field("format", formatEntries.get(0)),
+                            resultListener
+                        );
+                    }
+                } else {
+                    createEnrichMappingBuilder((builder) -> builder.field("type", matchType).field("doc_values", false), resultListener);
+                }
+            }
+        }
+    }
+
+    private List<String> collapseFieldMappingsValue(GetIndexResponse response, Function<Map<String, Object>, String> getPropertyFunction) {
+
+        return StreamSupport.stream(response.mappings().values().spliterator(), false)
+            .map(cursor -> cursor.value)
+            .map(MappingMetadata::sourceAsMap)
+            .map(map -> property(map, "properties"))
+            .map(getPropertyFunction)
+            .distinct()
+            .collect(Collectors.toList());
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<String, Object> property(Map<String, Object> map, String property) {
+        return (Map<String, Object>) map.getOrDefault(property, Map.of());
+    }
+
+    private Function<Map<String, Object>, String> toFieldMappingProperty(String matchField, String property) {
+        return map -> {
+            String[] parts = matchField.split("\\.");
+            Map<String, Object> result = map;
+            if (parts.length > 1) {
+                for (int i = 0; i < parts.length - 1; i++) {
+                    result = property(result, parts[i]);
+                    result = property(result, "properties");
+                }
+            }
+            result = property(result, parts[parts.length - 1]);
+            return String.valueOf(result.getOrDefault(property, "unknown"));
```

Comment:
I don't see handling for `unknown` as return value. Is this something that should be done? Or maybe just `null` should be returned?

Reply:
I think your right, I  did a pass to clean/simply things but am now no longer checking for unknown and for isEmpty instead. IT does still help with more human friendly error messages. Going to revisit this.

Reply:
adressed in https://github.com/elastic/elasticsearch/pull/76110/commits/44cab8e54cb59099114a07af78626c1130a74d09 / https://github.com/elastic/elasticsearch/pull/76110/commits/b6e71e8db37c15de85ebeb3173f82c0e023f447e